### PR TITLE
Pass props to toolbars and clean up config stores

### PR DESCRIPTION
**Pull request by axelboc**
:sparkles: :champagne: 

- Mapped vis no longer call `useEffect`.
- Everything stored in the toolbar config stores is now persistable/persisted.
- Some of the states like whether to disable the "errors" or "auto-scale" toggles are now just toolbar props, while other states like the initial scale types are now toolbar props combined with `useEffect` calls located in the toolbar components.

### CODE REVIEW

File: packages/app/src/vis-packs/core/complex/ComplexLineToolbar.tsx [hunk 1]
```diff
@@ -53,26 +71,24 @@ function ComplexLineToolbar() {
...
+        value={visType}
+        onChange={(value: ComplexLineVisType) => setVisType(value)}
+        options={[ComplexVisType.Amplitude, ComplexVisType.Phase]}
+        optionComponent={({ option }) => (
+          // eslint-disable-next-line react/jsx-no-useless-fragment
+          <>{`${VIS_TYPE_SYMBOLS[option]} ${option}`}</>
+        )}
+      />
+
+      <Separator />
```
Comment (axelboc) [inline packages/app/src/vis-packs/core/complex/ComplexLineToolbar.tsx:84]: I moved this up and added a separator.

File: packages/app/src/vis-packs/core/complex/ComplexLineToolbar.tsx [hunk 2]
```diff
@@ -53,26 +71,24 @@ function ComplexLineToolbar() {
...
-        icon={(props) => (
-          <FiItalic
-            transform="skewX(20)"
-            style={{ marginLeft: '-0.25em', marginRight: '0.0625rem' }}
-            {...props}
-          />
-        )}
-        value={!areErrorsDisabled && showErrors}
-        onToggle={toggleErrors}
-        disabled={areErrorsDisabled}
```
Comment (axelboc) [inline packages/app/src/vis-packs/core/complex/ComplexLineToolbar.tsx:75]: The NeXus complex line vis cannot show errors like the NeXus line vis can, so the button was always disabled.

File: packages/app/src/vis-packs/core/complex/ComplexToolbar.tsx [hunk 3]
```diff
@@ -83,6 +81,18 @@ function ComplexToolbar() {
...
+        value={visType}
+        onChange={(value: ComplexVisType) => setVisType(value)}
+        options={Object.values(ComplexVisType)}
+        optionComponent={({ option }) => (
+          // eslint-disable-next-line react/jsx-no-useless-fragment
+          <>{`${VIS_TYPE_SYMBOLS[option]} ${option}`}</>
+        )}
+      />
+
+      <Separator />
```
Comment (axelboc) [inline packages/app/src/vis-packs/core/complex/ComplexToolbar.tsx:94]: Moved this down for consistency with the complex line toolbars (scale type then vis type).

**Review by axelboc — COMMENTED** [review]
Might be easier to look at the line/heatmap vis before looking at the complex line/complex vis.

File: packages/app/src/vis-packs/core/complex/MappedComplexLineVis.tsx [hunk 4]
```diff
@@ -31,7 +32,14 @@ function MappedComplexLineVis(props: Props) {
   return (
     <>
       {toolbarContainer &&
-        createPortal(<ComplexLineToolbar />, toolbarContainer)}
+        createPortal(
+          <ComplexLineToolbar
+            initialXScaleType={axisMapping[dimMapping.indexOf('x')]?.scaleType}
+            initialYScaleType={valueScaleType}
+            withAutoScale={dims.length > 1} // with 1D datasets, `baseArray` and `dataArray` are the same
```
Comment (loichuder) [inline packages/app/src/vis-packs/core/complex/MappedComplexLineVis.tsx:39]: Why not keeping `disableAutoScale` ? `withAutoScale` makes me think that the button will disappear if set to `false`.

**Review by loichuder — APPROVED** [review]
Much smoother :relieved:

**Pull request merged**